swfiles/sw_egrid.m

```diff
@@ -450,20 +454,36 @@
 if eBin
     ebin_edges = param.Evect;
     ebin_cens = (ebin_edges(2:end)+ebin_edges(1:(end-1)))/2;
-    dE = diff(ebin_cens);
 else
     ebin_cens = param.Evect;
-    dE = diff(ebin_cens);
-    ebin_edges = [ebin_cens-0.5*[dE(1) dE], ebin_cens(end) + dE(end)/2];
+    den = diff(ebin_cens)./2;
+    ebin_edges = [ebin_cens(1)-den(1), ebin_cens(1:end-1) + den, ebin_cens(end) + den(end)];
 end
+bin_widths = diff(ebin_edges);
 nE = numel(ebin_cens);
 
+% evalulate reoslution at every bin-center if func
+if isa(param.dE,'function_handle')
+    dE_sigma = param.dE(ebin_cens(:))/2.355;
```

Comment:
Not sure if it matters but `2.35482` is accurate to 1e-7... (`2*sqrt(2*log(2)) = 2.3548200450...`)

Reply:
Might as well, sorry old habit!